k2/csrc/fsa_utils.cu

```diff
@@ -156,11 +162,12 @@ static Fsa K2AcceptorFromStream(std::istringstream &is) {
     }
   }
 
+  num_states = states_set.size();
```

Comment:
I am concerned about this line; remember, not all states may be seen, that would still be a valid FST.  (e.g. states 0,1,2,10,11,12 are seen).

Reply:
Should it be the following:
```cpp
num_states = largest state number in states_set + 1;
```
?

Reply:
If understood correctly, the only purpose we use `states_set` is to get the largest number?? why not just use a single int32_t variable (init with -1 and check >-1 finally) to do this.